File: app/src/androidTest/java/com/example/android/testing/espresso/BasicSample/Test1Test.java
package com.example.android.testing.espresso.BasicSample;

import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
import static android.support.test.espresso.action.ViewActions.typeText;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withText;

@RunWith(AndroidJUnit4.class)
public class Test1Test {

    public static final String STRING_TO_BE_TYPED = "Espresso";

    @Rule
    public ActivityTestRule<MainActivity> mActivityRule = new ActivityTestRule<>(
            MainActivity.class);


    @Test
    public void test1_newActivity_doubleText_error() {
        // Type text and then press the button.
        onView(withId(R.id.editTextUserInput)).perform(typeText(STRING_TO_BE_TYPED),
                closeSoftKeyboard());
        onView(withId(R.id.test1)).perform(click());

        // Error on this line ! but why ?
        onView(withText("Espresso")).check(matches(isDisplayed()));
    }

    // TODO make it right !
    @Test
    public void test1_newActivity_doubleText_passed() {
        
    }
}

File: app/src/androidTest/java/com/example/android/testing/espresso/basicSample/Test1Test.kt
package com.example.android.testing.espresso.basicSample

import androidx.test.rule.ActivityTestRule
import androidx.test.runner.AndroidJUnit4
import com.example.android.testing.espresso.basicSample.robots.MainRobot
import com.example.android.testing.espresso.basicSample.robots.main
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith


@RunWith(AndroidJUnit4::class)
class Test1Test {

    @get:Rule
    var mActivityRule: ActivityTestRule<MainActivity> = ActivityTestRule(
            MainActivity::class.java)

    @Test
    fun test1_newActivity_doubleText_error() {
        main {
            enterText("Espresso")
            clickTest1()
            hasText("Espresso")
        }
    }

    @Test
    fun test1_newActivity_doubleText_passed() {
        // Type text and then press the button.
        onView(withId(R.id.editTextUserInput)).perform(typeText("Espresso"),
                closeSoftKeyboard())
        onView(withId(R.id.test1)).perform(click())

        onView(withId(R.id.show_text_view_one)).check(matches(withText("Espresso")))
        onView(withId(R.id.show_text_view_two)).check(matches(withText("Espresso")))
    }
}